runtime/src/main/cpp/Alloc.h

```diff
@@ -119,4 +119,48 @@ bool operator!=(
     return !(x == y);
 }
 
+template <class T>
+class KonanDeleter {
+public:
+    void operator()(T* instance) noexcept { konanDestructInstance(instance); }
+};
+
+// Force a class to be heap-allocated using `konanAllocMemory`. Does not prevent stack allocation, or
+// allocation as part of another object.
+// Usage:
+// class A : public KonanAllocatorAware<A> {
+//     ...
+// };
+// To construct `A` on the heap:
+// new (A::Alloc()) A(...);
```

Comment:
Can we make it more like `new (KonanAlloc) A(...)`?
This might be even more simple and less fragile.

Reply:
That's much better, thanks!